packages/client/hmi-client/src/components/model/variables/tera-variable-entry.vue

```diff
@@ -1,18 +1,33 @@
 <template>
 	<section>
-		<h6>{{ variable.id }}</h6>
+		<h6>
+			<template v-if="variable.templateId">{{ variable.templateId }},</template> {{ variable.id }}
+		</h6>
 		<tera-input
 			title="Name"
 			placeholder="Add a name"
 			:model-value="variable.name ?? ''"
 			@update:model-value="$emit('update-variable', { key: 'name', value: $event })"
+			:disabled="disabledInputs?.includes('name')"
 		/>
+		<div v-if="variable.input && variable.output" label="Unit">
+			<span><span>Input:</span> {{ variable.input }}</span>
+			<span><span>Output:</span> {{ variable.output }}</span>
+		</div>
+		<!--amr_to_mmt doesn't like unit expressions with spaces, removing them here before they are saved to the amr-->
 		<tera-input
+			v-else
 			label="Unit"
 			placeholder="Add a unit"
 			:model-value="variable.unitExpression ?? ''"
-			@update:model-value="$emit('update-variable', { key: 'unitExpression', value: $event })"
+			@update:model-value="
+				($event) => {
+					const value = $event.replace(/[\s.]+/g, '');
+					$emit('update-variable', { key: 'unitExpression', value });
+				}
+			"
 			:disabled="disabledInputs?.includes('unitExpression')"
+			@focusout="($event) => ($event.target.value = $event.target.value.replace(/[\s.]+/g, ''))"
```

Comment:
- Let's try to keep the template section clean, you are doing too much data wrangling here
- If MIRA is having problem parsing spaces, that is issue that should be raised and discussed before we try to work around it

Reply:
Yeah the `amr_to_mmt` doesn't work if there is a space in the unit expression. I'll remove the string edit. How would it handle spaces though, would they still be there? I think the UI should reflect what's in MIRA whether it removes the space or replaces it with a `_`

Reply:
We can keep the space truncation hack for now. 

But we should find and examine what the error actually is, because you never know if it is just a symptom of a deeper problem. Basically if other performer's stuff error out, the error should always be raised for awareness and triaging.

Do you have a couple of examples of the strings you had tried and failed? And what does the the stack trace look like? Let's take the conversation off this PR and into our slack channel.